### ISSUE TITLE
'popScope' has different behaivior on develop / release build.  

### ISSUE THREAD

**Issue by swissonid** [fluttercommunity/get_it#152]
I do not know if its intentional or not. The assertion gets removed/disabled during the release build. Which can lead to strange behaviour without any meaningful error message. Or just not to pop if you are on the baseScope

```dart  
@override
  Future<void> popScope() async {
    assert(
        _scopes.length > 1,
        "You are already on the base scope. you can't pop"
        ' this one');
    await _currentScope.dispose();
    await _currentScope.reset(dispose: true);
    _scopes.removeLast();
  }
```

**Comment by escamoteur** [comment]
That is correct. But that's the idea of asserts in general. Do you see it as an error situation that could happen in release that wouldn't be a fatal error that should have been detected during development?

**Comment by escamoteur** [comment]
yes, IMHO if that happens something is wrong with your App, so it's better to have an assert.

**Comment by swissonid** [comment]
Ok, let's assume something is wrong in my app. Still would it not be nice if GetIt would prevent the User, in that case, the developer to make so silly mistake as I did? So that the end user gets a better experience. And print a warning or event throw an exception?

**Comment by escamoteur** [comment]
I'm a big fan of ailing fast :-) But I agree we could change this into an if with an exception instead of the assert, so that you get a meaningful error message

**Comment by escamoteur** [comment]
Implemented in V6.1.0
